docs/architecture-decisions/0005-implement-an-accessible-navigation-system.md

```diff
@@ -0,0 +1,202 @@
+  by stylesheets alone. An example is [WCAG 1.4.13](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html), which requires that revealed
+  content like a submenu must be dismissable without changing the focus on a
+  page.
+
+
+### Option 4: A bespoke solution
+
+While attempting to implement the Accessible Menu library, we asked for feedback
+about our progress from Rich Caloggero, who was involved in our earlier
+accessibility testing. Rich helpfully assembled a [mockup](https://richcaloggero.github.io/tools/createNavigation.html) demonstrating a usable
+navigation system with very little javascript (and no stylesheets).
+
+#### Benefits of a bespoke approach
+
+The amount of javascript which is needed is potentially very, very small. The
+initial snippet which Rich provided was just this:
+
+```js
+$submenus = document.querySelectorAll("nav ul ul");
+$submenus.forEach(x => x.setAttribute("hidden", ""));
+$nav.removeEventListener("click", clickHandler);
+$nav.addEventListener("click", clickHandler);
+
+function clickHandler (e) {
+if (e.target instanceof HTMLButtonElement) {
+e.preventDefault();
+const menu = e.target.parentElement instanceof HTMLLIElement? e.target.nextElementSibling  : e.target.parentElement.nextElementSibling;
+menu.hidden = not(menu.hidden);
+e.target.setAttribute("aria-expanded", menu.hidden? "false" : "true");
+} // if
+} // clickHandler
+```
+
+If we choose HTML elements which bring native functionality (`button`, `a`, and
+header tags, for example) then we not only need less javascript but we are more
+likely to match functionality which users are already familiar with using. The
+only conceptual need we are solving is how to expand and collapse a submenu -
+everything else is already present.
+
+The styling rules required for this approach should likewise be minimal.
+
+#### Drawbacks of a bespoke approach
+
+The maxim "if you aren't using a framework, you are building one" applies here.
+While the amount of code required is small, it is still code for which we are
+responsible. That code will require us to commit to maintaining the knowledge of
+how it works and how to troubleshoot it, and to commit to having the bandwidth
+to respond to problems as they are reported.
+
+This bespoke approach is also not just javascript, but all the associated
+styling rules for the menu to be rendered in both a vertical and horizontal
+layout.
+
+In addition, because this code is so small, it has likely been written by some
+other group of developers - and it would be preferable to leverage their work
+rather than repeat it.
+
+Ultimately, though, the review which we've conducted of existing menu libraries
+has not identified such a suitable existing library at this time.
+
+
+## Decision
+
+We will develop, and maintain, a small set of javascript and stylesheet rules
```

Comment:
Can you explicitly name the option we are choosing. i.e. "We will implement `Option 4: A bespoke solution. We will develop..."